apps/api/src/app/workflows-v2/usecases/generate-preview/generate-preview.usecase.ts

```diff
@@ -41,10 +53,55 @@ export class GeneratePreviewUsecase {
         preview: executeOutput.outputs as any,
         type: stepData.type as unknown as ChannelTypeEnum,
       },
-      previewPayloadExample: validatedContent.finalPayload,
+      previewPayloadExample: hydratedExampleVariables,
     };
   }
 
+  private async findWorkflow(command: GeneratePreviewCommand) {
+    return await this.getWorkflowByIdsUseCase.execute(
+      GetWorkflowByIdsCommand.create({
+        identifierOrInternalId: command.identifierOrInternalId,
+        environmentId: command.user.environmentId,
+        organizationId: command.user.organizationId,
+        userId: command.user._id,
+      })
+    );
+  }
+
+  /**
+   * Generates a payload based solely on the schema.
+   * Supports nested schemas and applies defaults where defined.
+   * @param JSONSchemaDto - Defining the structure. example:
+   *  {
+   *    firstName: { type: 'string', default: 'John' },
+   *    lastName: { type: 'string' }
+   *  }
+   * @returns - Generated payload. example: { firstName: 'John', lastName: '{{payload.lastName}}' }
+   */
+  private generatePayloadVariableExample(schema: JSONSchemaDto, path = 'payload'): Record<string, unknown> {
+    if (schema.type !== 'object' || !schema.properties) {
+      throw new Error('Schema must define an object with properties.');
+    }
+
+    return Object.entries(schema.properties).reduce((acc, [key, definition]) => {
+      if (typeof definition === 'boolean') {
+        return acc;
+      }
+
+      const currentPath = `${path}.${key}`;
+
+      if (definition.default) {
+        acc[key] = definition.default;
+      } else if (definition.type === 'object' && definition.properties) {
+        acc[key] = this.generatePayloadVariableExample(definition, currentPath);
```

Comment:
Please add a TODO here to add a maximum number of iterations to avoid an infinite loop. These schemas come from code, so we need to prevent very deep structures to avoid resource exhaustion attacks.

Reply:
done added max depth of 10, i belive it will be enough for now.